pkg/parser/document_processing_apply_substitutions.go

```diff
@@ -74,6 +76,48 @@ func applyAttributeSubstitutions(elements []interface{}, attrs types.AttributesW
 
 }
 
+func applyCounterSubstitution(c types.CounterSubstitution, attrs types.AttributesWithOverrides) (interface{}, bool, error) {
```

Comment:
could you add some comments on this func to explain the returned values. Also, Do you really need to return a `bool`? AFAICS, it's always `true` unless there is an error.

Reply:
I can.  I think I used the same context for bool as ApplySubsitutions did -- I wanted this to conform to the same signature.

Reply:
Basically it has to conform to applyAttributeSubstitutionsOnElement.

Reply:
Comment applied -- I will squash if you approve.

Reply:
ok, I see. Thanks for the clarification.